code/Controllers/CMSMain.php

```diff
@@ -2064,12 +2042,8 @@ public function rollback()
 
     /**
      * Rolls a site back to a given version ID
-     *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
      */
-    public function doRollback($data, $form)
+    public function doRollback(array $data, Form $form): HTTPResponse
```

Comment:
Also this method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array